components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/provider/ApplicationRoleProvider.java

```diff
@@ -0,0 +1,18 @@
+package org.wso2.carbon.identity.application.mgt.provider;
+
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+
+/**
+ * Interface to provide application roles.
+ */
+public interface ApplicationRoleProvider {
+
+    /**
+     * Get user roles.
+     *
+     * @param username Username.
+     * @param tenantId Tenant ID.
+     * @return Array of user roles.
+     */
+    String[] getUserRoles(String username, int tenantId) throws IdentityApplicationManagementException;
```

Comment:
the method name and the class name is confusing. shouldn't this be just getRoles(), and have application, creator, tenant as the params/context?

Reply:
Fixed. Change the method signature as above